pkg/apiserver/api_installer.go

```diff
@@ -379,26 +384,32 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 				Filter(m).
 				Doc("partially update the specified "+kind).
 				Consumes(string(api.JSONPatchType), string(api.MergePatchType), string(api.StrategicMergePatchType)).
-				Operation("patch" + kind).
+				Operation("patch"+kind).
 				Produces(append(storageMeta.ProducesMIMETypes(action.Verb), "application/json")...).
-				Reads(versionedObject)
+				Returns(http.StatusOK, "OK", "string").
+				Reads(versionedObject).
+				Writes("string")
```

Comment:
Does passing `"string"` actually work, because `"string"` happens to be a string? 

Hmm. It doesn't look like the re-generated swagger has changed at all due to your changes.

Anyway, this is the opposite of what I meant.

Patch should read a patch specification, which might be one of 3 different forms, and should return the versioned object. So, it should be:
```
Reads("string").
Writes(versionedObject).
Returns(http.StatusOK, "OK", versionedObject)
```